File: algorithms/recursion/sigma/recursive_sigma.js
function r_sigma(num) {
  if (num <= 0) {
    return "Invalid number";
  }
  if (num == 1) {
    return 1;
  }
  return num + r_sigma(num-1);
}

console.log(r_sigma(5));
console.log(r_sigma(1));
console.log(r_sigma(0));
console.log(r_sigma(2));
console.log(r_sigma(10));

File: algorithms/recursion/sigma/recursive_sigma.rb
def r_sigma(num)
  if num <= 0
    return "Invalid Number"
  end
  if num == 1
    return 1
  end
  return num + r_sigma(num-1)
end

p(r_sigma(5))
p(r_sigma(1))
p(r_sigma(0))
p(r_sigma(2))
p(r_sigma(10))
